### PR TITLE
Added "hasAllNullFieldsOrProperties" and "hasAllNullFieldsOrPropertiesExcept"

### LINKED ISSUE #1305: Please add "hasAllNullFieldsOrProperties" and "hasAllNullFieldsOrPropertiesExcept"
#### Summary
Please add `hasAllNullFieldsOrProperties` as opposite to `hasNoNullFieldsOrProperties`
and `hasAllNullFieldsOrPropertiesExcept` as opposite to `hasNoNullFieldsOrPropertiesExcept`

### PR DESCRIPTION
**Pull request by Mef1STo**
#### Check List:
* Fixes #1305
* Unit tests : YES
* Javadoc with a code example (API only) : YES

### CODE REVIEW

File: src/main/java/org/assertj/core/api/AbstractObjectAssert.java [hunk 1]
```diff
@@ -226,24 +225,54 @@ public SELF hasNoNullFieldsOrProperties() {
     return myself;
   }
 
+  /**
+   * Asserts that the actual object has null fields or properties.
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/api/AbstractObjectAssert.java]: this is not an accurate description of the assertion

File: src/main/java/org/assertj/core/api/AbstractObjectAssert.java [hunk 2]
```diff
@@ -257,6 +286,37 @@ public SELF hasNoNullFieldsOrPropertiesExcept(String... propertiesOrFieldsToIgno
     return myself;
   }
 
+  /**
+   * Asserts that the actual object has null fields or properties <b>except for the given ones</b>
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/api/AbstractObjectAssert.java]: replace `has null` by `has only null`

File: src/main/java/org/assertj/core/error/ShouldHaveAllNullFields.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+import java.util.List;
+
+/**
+ * Creates an <code>{@link AssertionError}</code> indicating that an assertion that verifies that an object
+ * has null fields failed.
+ *
+ * @author Vladimir Chernikov
+ */
+public class ShouldHaveAllNullFields extends BasicErrorMessageFactory {
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/error/ShouldHaveAllNullFields.java]: add unit tests ina class named `ShouldHaveAllNullFields_create_Test`

File: src/main/java/org/assertj/core/internal/Objects.java [hunk 4]
```diff
@@ -709,6 +711,39 @@ static boolean propertyOrFieldValuesAreEqual(Object actualFieldValue, Object oth
...
+        continue;
+      }
+      Object actualFieldValue = getPropertyOrFieldValue(actual, fieldName);
+      if (actualFieldValue != null) {
+        notNullFieldNames.add(fieldName);
+      }
+    }
+    if (!notNullFieldNames.isEmpty()) {
+      throw failures.failure(info, shouldHaveAllNullFields(
+        actual, notNullFieldNames, newArrayList(propertiesOrFieldsToIgnore)
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/internal/Objects.java]: use `list` in place of `newArrayList`

File: src/main/java/org/assertj/core/error/ShouldHaveAllNullFields.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  private static final String EXPECTED_SINGLE = "%nExpecting%n  <%s>%nto have null property or field, but <%s> was not null.%n";
+  private static final String COMPARISON = "Check was performed on all fields/properties";
+  private static final String EXCLUDING = COMPARISON + " except: <%s>.";
+  private static final String DOT = ".";
+
+  public ShouldHaveAllNullFields(Object actual, List<String> rejectedFields, List<String> ignoredFields) {
+    super(EXPECTED_MULTIPLE + EXCLUDING, actual, rejectedFields, ignoredFields);
+  }
+
+  public ShouldHaveAllNullFields(Object actual, List<String> rejectedFields) {
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/error/ShouldHaveAllNullFields.java]: rename `rejectedFields` to `nonNullFields` (everywhere) as it better describes what is the parameter for.

File: src/main/java/org/assertj/core/error/ShouldHaveAllNullFields.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+/**
+ * Creates an <code>{@link AssertionError}</code> indicating that an assertion that verifies that an object
+ * has null fields failed.
+ *
+ * @author Vladimir Chernikov
+ */
+public class ShouldHaveAllNullFields extends BasicErrorMessageFactory {
+
+  private static final String EXPECTED_MULTIPLE = "%nExpecting%n  <%s>%nto have properties or fields named <%s> are null.%n";
+  private static final String EXPECTED_SINGLE = "%nExpecting%n  <%s>%nto have null property or field, but <%s> was not null.%n";
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/error/ShouldHaveAllNullFields.java]: replace `to have null property or field` by `to only have null properties or fields`

File: src/main/java/org/assertj/core/error/ShouldHaveAllNullFields.java [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+/**
+ * Creates an <code>{@link AssertionError}</code> indicating that an assertion that verifies that an object
+ * has null fields failed.
+ *
+ * @author Vladimir Chernikov
+ */
+public class ShouldHaveAllNullFields extends BasicErrorMessageFactory {
+
+  private static final String EXPECTED_MULTIPLE = "%nExpecting%n  <%s>%nto have properties or fields named <%s> are null.%n";
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/error/ShouldHaveAllNullFields.java]: replace by `"%nExpecting%n  <%s>%nto only have null properties or fields but these were not null:%n  <%s> are null.%n";`

**Comment by joel-costigliola** [comment]
Thanks @Mef1STo! I have made a first review of your PR.

File: src/main/java/org/assertj/core/api/AbstractObjectAssert.java (hunk 1 above)
Comment (Mef1STo) [reply src/main/java/org/assertj/core/api/AbstractObjectAssert.java]: What do you exactly mean? 
``` Asserts that the actual object has only null fields or properties. ```
Is it fine? Or did you mean something else?

File: src/main/java/org/assertj/core/api/AbstractObjectAssert.java (hunk 2 above)
Comment (Mef1STo) [reply src/main/java/org/assertj/core/api/AbstractObjectAssert.java]: Changed.

File: src/main/java/org/assertj/core/error/ShouldHaveAllNullFields.java (hunk 3 above)
Comment (Mef1STo) [reply src/main/java/org/assertj/core/error/ShouldHaveAllNullFields.java]: Tests were added.

File: src/main/java/org/assertj/core/internal/Objects.java (hunk 4 above)
Comment (Mef1STo) [reply src/main/java/org/assertj/core/internal/Objects.java]: Fixed.

File: src/main/java/org/assertj/core/error/ShouldHaveAllNullFields.java (hunk 5 above)
Comment (Mef1STo) [reply src/main/java/org/assertj/core/error/ShouldHaveAllNullFields.java]: Renamed.

File: src/main/java/org/assertj/core/error/ShouldHaveAllNullFields.java (hunk 6 above)
Comment (Mef1STo) [reply src/main/java/org/assertj/core/error/ShouldHaveAllNullFields.java]: Changed.

File: src/main/java/org/assertj/core/error/ShouldHaveAllNullFields.java (hunk 7 above)
Comment (Mef1STo) [reply src/main/java/org/assertj/core/error/ShouldHaveAllNullFields.java]: Changed.

**Comment by Mef1STo** [comment]
Hm... build failed with
```
[ERROR] Tests run: 7, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 0.039 s <<< FAILURE! - in org.assertj.core.api.objectarray.ObjectArrayAssert_extractingResultOf_Test
[ERROR] extractingResultOf_should_keep_assertion_state  Time elapsed: 0.034 s  <<< ERROR!
org.assertj.core.util.introspection.IntrospectionError: 
Can't find any field or property with name 'comparatorsByType'.
Error when introspecting properties was :
- No public getter for property 'comparatorsByType' in org.assertj.core.api.ListAssert 
Error when introspecting fields was :
- Unable to obtain the value of the field <'comparatorsByType'> from <org.assertj.core.api.ListAssert@1>, check that field is public.
	at org.assertj.core.api.objectarray.ObjectArrayAssert_extractingResultOf_Test.extractingResultOf_should_keep_assertion_state(ObjectArrayAssert_extractingResultOf_Test.java:115)
Caused by: org.assertj.core.util.introspection.IntrospectionError: Unable to obtain the value of the field <'comparatorsByType'> from <org.assertj.core.api.ListAssert@1>, check that field is public.
	at org.assertj.core.api.objectarray.ObjectArrayAssert_extractingResultOf_Test.extractingResultOf_should_keep_assertion_state(ObjectArrayAssert_extractingResultOf_Test.java:115)
Caused by: java.lang.IllegalAccessException: can not accesscomparatorsByType because it is not public
	at org.assertj.core.api.objectarray.ObjectArrayAssert_extractingResultOf_Test.extractingResultOf_should_keep_assertion_state(ObjectArrayAssert_extractingResultOf_Test.java:115)
```
Have any idea? Locally I cannot reproduce it.

**Comment by joel-costigliola** [comment]
it is also failing in master, not sure why though since it is working locally for me.
I believe it might due to some update in Travis

**Comment by joel-costigliola** [comment]
Integrated thanks @Mef1STo !

**Pull request closed without merging**